woocommerce/woocommerce
# Manual coupons support

Stores data about manual coupons so they can be re-applied in admin.

`woocommerce_order_recalculate_coupons_coupon_object` filter can tweak coupons before they are reapplied if more advanced code is needed.

`is_virtual` method added to coupon objects to track if they are manual or not.

To test:

1. Add manual coupon to cart
2. Add regular coupon to cart
3. Place order
4. Remove regular coupon from order
5. Check discounts are applied and manual coupon has correct discount amount set.

Fixes #16594
